Userland/Libraries/LibJIT/GDB.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2023, Jesús Lapastora <cyber.gsuscode@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Checked.h>
+#include <AK/OwnPtr.h>
+#include <AK/Span.h>
+#include <LibJIT/GDB.h>
+
+namespace JIT::GDB {
+
+// Declarations from https://sourceware.org/gdb/current/onlinedocs/gdb.html/Declarations.html#Declarations.
+// NOTE: If the JIT is multi-threaded, then it is important that the JIT synchronize any modifications to this global data properly, which can easily be done by putting a global mutex around modifications to these structures.
+extern "C" {
+typedef enum {
+    JIT_NOACTION = 0,
+    JIT_REGISTER_FN,
+    JIT_UNREGISTER_FN
+} jit_actions_t;
+
+struct jit_code_entry {
+    struct jit_code_entry* next_entry;
+    struct jit_code_entry* prev_entry;
+    char const* symfile_addr;
+    u64 symfile_size;
+};
+
+struct jit_descriptor {
+    u32 version;
+    // This type should be jit_actions_t, but we use uint32_t to be explicit
+    // about the bitwidth.
+    u32 action_flag;
+    struct jit_code_entry* relevant_entry;
+    struct jit_code_entry* first_entry;
+};
```

Comment:
The structure names don't matter for the ABI, so I think these should be changed to match our coding style.